.github/workflows/publish-official-release.yaml

```
@@ -0,0 +1,185 @@
+
+      - name: Checkout Code
+        uses: actions/checkout@11bd71901bbe5b1630ceea73d27597364c9af683 # v4.2.2
+        with:
+          token: ${{ secrets.GH_ACCESS_TOKEN }}
+          fetch-depth: 0
+          ref: ${{ inputs.branch }}
+
+      - name: Setup Java
+        if: ${{ inputs.dry-run-enabled != true && !cancelled() && !failure() }}
+        uses: actions/setup-java@7a6d8a8234af8eb26422e24e3006232cccaa061b # v4.6.0
+        with:
+          distribution: ${{ inputs.java-distribution || 'temurin' }}
+          java-version: ${{ inputs.java-version || '21.0.4' }}
+
+      - name: Setup Gradle
+        if: ${{ inputs.dry-run-enabled != true && !cancelled() && !failure() }}
+        uses: gradle/actions/setup-gradle@0bdd871935719febd78681f197cd39af5b6e16a6 # v4.2.2
+        with:
+          gradle-version: wrapper
+
+      # This step is required for some self-hosted runners because the default gpg does not include the gpg2 command.
+      - name: Install GnuPG Tools
+        run: |
+          if ! command -v gpg2 >/dev/null 2>&1; then
+            echo "::group::Updating APT Repository Indices"
+              sudo apt update
+            echo "::endgroup::"
+            echo "::group::Installing GnuPG Tools"
+              sudo apt install -y gnupg2
+            echo "::endgroup::"
+          fi
+
+      - name: Import GPG Key
+        id: gpg
+        uses: step-security/ghaction-import-gpg@6c8fe4d0126a59d57c21f87c9ae5dd3451fa3cca # v6.1.0
+        with:
+          git_commit_gpgsign: true
+          git_tag_gpgsign: true
+          git_user_signingkey: true
+          gpg_private_key: ${{ secrets.GPG_KEY_CONTENTS }}
+          passphrase: ${{ secrets.GPG_KEY_PASSPHRASE }}
+
+      - name: Setup Node
+        uses: actions/setup-node@39370e3970a6d050c480ffad4ff0ed4d3fdee5af # v4.1.0
+        with:
+          node-version: 20
+
+      - name: Install Dependencies
+        run: |
+          echo "Installing semantic-release"
+          npm install
+
+      # Dry run will only output logs which can be used to verify the release process and next version
+      # Actual release will also publish a tag and release and update version.txt and CHANGELOG.md
+      - name: Run Semantic Release
+        if: ${{ !cancelled() && !failure() }}
+        env:
+          GH_TOKEN: ${{ secrets.GH_ACCESS_TOKEN }}
+          GIT_AUTHOR_NAME: ${{ steps.gpg.outputs.name}}
+          GIT_AUTHOR_EMAIL: ${{ steps.gpg.outputs.email}}
+          GIT_COMMITTER_NAME: ${{ steps.gpg.outputs.name}}
+          GIT_COMMITTER_EMAIL: ${{ steps.gpg.outputs.email}}
+          # The EDITOR var is required to ensure the tag is signed with the GPG key
+          # as semantic-release doesn't support headless annotated signed tags and
+          # the process hangs indefinitely awaiting input in the dialogue
+          EDITOR: "echo 'Created-by: semantic-release' > $1"
+        run: |
+          if [[ "${{ inputs.dry-run-enabled }}" == true ]]; then
```

does this pass? It looks like it's a string compared to a boolean value. I think i would expect it to be `[[ "${{ inputs.dry-run-enabled }}" == "true" ]]`

No change required just verifying this part has been tested.

---

Yes, tested in latest dry-run test https://github.com/hashgraph/hedera-cryptography/actions/runs/13306659441/job/37159090949

---

```suggestion
          if [[ "${{ inputs.dry-run-enabled }}" == "true" ]]; then
```

Okay, I think we should do this.

---

Okay, I'll add this commit and do a test dry run

---

Tested and it looks good - https://github.com/hashgraph/hedera-cryptography/actions/runs/13671461633/job/38222413717